toolkit/proto/protos/to_domain.proto

```diff
@@ -257,6 +257,27 @@ message WrapPrivacyGroupEVMTXResponse {
   PreparedTransaction transaction = 1; // The transaction that will result from this against the domain
 }
 
+message UnavailableStates {
+  optional string first_unavailable_id = 1; // convenience check - set in the order info, input, output, read
+  repeated string input_state_ids = 2; // unavailable input state ids
+  repeated string read_state_ids = 3; // unavailable read state ids
+  repeated string output_state_ids = 4; // unavailable output state ids
+  repeated string info_state_ids = 5; // unavailable info state ids
+}
+
+message CheckStateCompletionRequest {
+  string transaction_id = 1; // the transaction ID
+  repeated EndorsableState input_states = 2; // available input states
+  repeated EndorsableState read_states = 3; // available read states
+  repeated EndorsableState output_states = 4; // available output states
+  repeated EndorsableState info_states = 5; // available info states
+  UnavailableStates unavailable_states = 6; // details of unavailable states by classification, including a could for a quick completion check
+}
+
+message CheckStateCompletionResponse {
+  optional string primary_missing_state_id = 1; // if set, then this transaction is considered incomplete until this state is available - at which point the completion check will run again with the new set of states available at that time
```

Comment:
Is there anything that makes one state more "primary" than others? Is this "I'm missing at least one state" or "I'm missing _the primary_ state"?